scipy/_lib/_array_api.py

```diff
@@ -90,7 +98,7 @@ def array_namespace(*arrays):
 
     arrays = [array for array in arrays if array is not None]
 
-    compliance_scipy(*arrays)
+    arrays = compliance_scipy(*arrays)
```

Comment:
Since `arrays` is already a list, you save some lines and avoid repeated tuple concatenation by passing `arrays` instead of `*arrays` and modifying the elements in place (i.e. loop over the indices rather than elements).